test/tests_for_sdkmr/job_submission_parameters_test.py

```diff
@@ -175,22 +183,35 @@ def test_equals():
     assert JobRequirements(1, 1, 1, c1) != JobRequirements(1, 1, 1, c2)
     assert JobRequirements(1, 1, 1, c1) != (1, 1, 1, c1)
 
-    assert JobRequirements(1, 1, 1, c1, t, u1, f, r1) != JobRequirements(
-        1, 1, 1, c1a, f, u1a, f, r1a
+    assert JobRequirements(1, 1, 1, c1, t, u1, f, r1, t) != JobRequirements(
```

Comment:
If you're using the same JobRequirements object as the first comparator in each case, it would be easier to save it (or the params) to a variable and reuse that.

I'd also sort by the first comparator.

Reply:
> I'd also sort by the first comparator, then by the second comparator, so it's fairly obvious how you've progressed through altering the params to test inequality.

I don't think I understand what you're asking for here - it's sorted in order of the changing input, except for inserting testing against a tuple vs a class for required inputs and all inputs

Reply:
1st paragraph done

Reply:
As long as they are in an understandable order, it doesn't matter too much. They look fine now.

Reply:
It's a lot easier to see what's going on when everything's on one line for sure